packages/auto-label/src/auto-label.ts

```diff
@@ -211,66 +240,175 @@ handler.autoDetectLabel = (
+        autoDetectedLabel
+      );
+    } else {
+      console.log(
+        `Auto-detect did not add a label for ${owner}/${repo}#${issueNumber}: '${context.payload.issue.title}'`
+      );
+    }
+    return;
+  }
+
+  const colorNumber = jsonArray.findIndex(
+    (object: JSONData) => objectInJsonArray === object
+  );
+  const githubLabel = objectInJsonArray.github_label;
+
+  const alreadyExists = await handler.checkExistingLabels(
+    context.github,
+    owner,
+    repo,
+    githubLabel
+  );
+
+  if (alreadyExists === null || alreadyExists === undefined) {
+    handler.createLabel(
+      context.github,
+      owner,
+      repo,
+      githubLabel,
+      colorsData[colorNumber].color
+    );
+  } else {
+    console.log(
+      'This label already exists on the repository, will check if it also exists on the issue'
+    );
+  }
+
+  if (labelsOnIssue) {
+    const found = labelsOnIssue.find(
+      (element: {name: string}) => element.name === githubLabel
+    );
+    const cleanUpOtherLabels = labelsOnIssue.filter(
+      (element: {name: string}) =>
+        element.name.startsWith('api') &&
+        element.name !== found?.name &&
+        element.name !== autoDetectedLabelAdded
+    );
+    if (found) {
+      console.log('This label already exists on this issue');
+      return;
+    } else {
+      await handler.addLabels(context.github, owner, repo, issueNumber, [
+        `${objectInJsonArray.github_label}`,
+      ]);
+    }
+    for (const dirtyLabel of cleanUpOtherLabels) {
+      await handler.removeLabel(
+        context.github,
+        owner,
+        repo,
+        issueNumber,
+        dirtyLabel.name
+      );
+    }
+  } else {
+    await handler.addLabels(context.github, owner, repo, issueNumber, [
+      `${objectInJsonArray.github_label}`,
+    ]);
+  }
+
+  return;
+};
+
+//main function, responds to label being added
 function handler(app: Application) {
-  app.on(['issues.labeled'], async context => {
-    const owner = context.payload.repository.owner.login;
+  //nightly cron that backfills and corrects api labels
+  app.on(['schedule.repository'], async context => {
+    console.info(`running for org ${context.payload.cron_org}`);
+    const owner = context.payload.organization.login;
     const repo = context.payload.repository.name;
-    // if missing the label, skip
+    if (context.payload.cron_org !== owner) {
+      console.log(`skipping run for ${context.payload.cron_org}`);
+      return;
+    }
     if (
-      !context.payload.issue.labels.some(
-        (label: {name: string}) => label.name === BACKFILL_LABEL
+      repos.includes(
+        `${context.payload.organization.login}/${context.payload.repository.name}`
       )
     ) {
-      app.log.info(
-        `ignoring non-backfill label action (${context.payload.issue.labels.join(
-          ', '
-        )})`
-      );
       return;
     }
 
+    console.log(path.join(__dirname, './repos.txt'));
+    fs.writeFileSync(
```

Comment:
This seems like it writes to the Cloud Function's local file system? The file system is not shared between Cloud Functions instances, so I don't think this will work.

It would be OK with me to run over every issue every time if we know that it's correct.

Reply:
I was worried about that... thanks for confirming!

Well, to your point, it's every issue for every one of our repos.

Is it worthwhile/are there any disadvantages to writing these repos to Datastore? My only concerns here are that (i) I feel like this solution would just replicate repos.json in sloth (which is something we're trying to eliminate), and (ii), a repo may change a product overview without actually changing its title, in which case this cron job wouldn't run for it.